### PR TITLE
Ag setup devise

### PR DESCRIPTION
**Pull request by jasonramirez**
screen shot 2015-02-18 at 11 33 31 am
screen shot 2015-02-18 at 11 12 20 am
screen shot 2015-02-18 at 11 12 23 am
screen shot 2015-02-18 at 11 12 14 am

### CODE REVIEW

File: app/views/pages/home.html.slim [hunk 1]
```diff
@@ -1,3 +1,10 @@
+nav
+  - if user_signed_in?
+    == link_to"Sign out", destroy_user_session_path, method: :delete
```
Comment (Euraldius) [inline app/views/pages/home.html.slim]: Space between `link_to` and `"Sign out"`

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,260 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
```
Comment (Euraldius) [inline config/initializers/devise.rb]: Let's kill all of these comments

File: app/views/users/shared/_links.html.slim [hunk 3]
```diff
@@ -0,0 +1,19 @@
+- if controller_name != 'sessions'
```
Comment (Euraldius) [inline app/views/users/shared/_links.html.slim]: Can we remove this file, and put the links where we need them? We definitely don't need all this logic at this point

Comment (jasonramirez) [inline app/views/users/shared/_links.html.slim]: Pared it down a bit. Seems like some of the logic might be useful

File: db/migrate/20150216201240_devise_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      t.datetime :remember_created_at
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.inet     :current_sign_in_ip
+      t.inet     :last_sign_in_ip
+
+      ## Confirmable
```
Comment (jasonramirez) [inline db/migrate/20150216201240_devise_create_users.rb]: Remove these comments too?

File: app/views/users/registrations/new.html.slim [hunk 5]
```diff
@@ -0,0 +1,13 @@
+section class="section"
+  div class="section__container"
+    h2
+      | Sign up dude
```
Comment (jasonramirez) [inline app/views/users/registrations/new.html.slim]: oops. Removing dude.

File: db/migrate/20150216201240_devise_create_users.rb (hunk 4 above)
Comment (Euraldius) [inline db/migrate/20150216201240_devise_create_users.rb]: Might as well

File: app/views/users/passwords/edit.html.slim [hunk 6]
```diff
@@ -0,0 +1,14 @@
+section class="section"
+  div class="section__container"
+    h2
+      | Change your password
+    = simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f|
+      = f.error_notification
+      = f.input :reset_password_token, as: :hidden
+      = f.full_error :reset_password_token
+      .form-inputs
+        = f.input :password, label: "New password", required: true, autofocus: true
```
Comment (Euraldius) [inline app/views/users/passwords/edit.html.slim]: Should these labels use `i18n`?

File: app/views/users/passwords/new.html.slim [hunk 7]
```diff
@@ -0,0 +1,11 @@
+section class="section"
+  div class="section__container"
+    h2
+      | Forgot your password?
+    = simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f|
+      = f.error_notification
+      .form-inputs
+        = f.input :email, required: true, autofocus: true
+      .form-actions
+        = f.button :submit, "Send me reset password instructions"
```
Comment (Euraldius) [inline app/views/users/passwords/new.html.slim]: More `i18n`

File: app/views/users/registrations/edit.html.slim [hunk 8]
```diff
@@ -0,0 +1,24 @@
+section class="section"
```
Comment (Euraldius) [inline app/views/users/registrations/edit.html.slim]: Can there be some whitespace added in this file to make it more readable?

File: app/views/users/registrations/edit.html.slim [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        = f.input :password, autocomplete: "off", hint: "leave it blank if you don't want to change it", required: false
+        = f.input :password_confirmation, required: false
+        = f.input :current_password, hint: "we need your current password to confirm your changes", required: true
+      .form-actions
+        = f.button :submit, "Update"
+    h3
+      | Cancel my account
+    p
+      | Unhappy? 
+      = link_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete
```
Comment (Euraldius) [inline app/views/users/registrations/edit.html.slim]: `i18n`

File: app/views/users/registrations/new.html.slim [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    h2
+      | Sign up
+    = simple_form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f|
+      = f.error_notification
+      .form-inputs
+        = f.input :email, required: true, autofocus: true
+        = f.input :password, required: true, hint: ("#{@minimum_password_length} characters minimum" if @validatable)
+        = f.input :password_confirmation, required: true
+      .form-actions
+        = f.button :submit, "Sign up"
```
Comment (Euraldius) [inline app/views/users/registrations/new.html.slim]: `i18n`

File: app/views/users/sessions/new.html.slim [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  div class="section__container"
+    h2
+      | Log in
+    = simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f|
+      .form-inputs
+        = f.input :email, required: false, autofocus: true
+        = f.input :password, required: false
+        = f.input :remember_me, as: :boolean if devise_mapping.rememberable?
+      .form-actions
+        = f.button :submit, "Log in"
```
Comment (Euraldius) [inline app/views/users/sessions/new.html.slim]: `i18n`

File: app/views/users/shared/_links.html.slim [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+  = link_to "Log in", new_session_path(resource_name)
+  br
+- if devise_mapping.registerable? && controller_name != 'registrations'
+  = link_to "Sign up", new_registration_path(resource_name)
+  br
+- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations'
+  = link_to "Forgot your password?", new_password_path(resource_name)
```
Comment (Euraldius) [inline app/views/users/shared/_links.html.slim]: `i18n` this file

File: config/initializers/devise.rb [hunk 13]
```diff
@@ -0,0 +1,18 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  config.scoped_views = true
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  require 'devise/orm/active_record'
+
+  config.case_insensitive_keys = [ :email ]
```
Comment (Euraldius) [inline config/initializers/devise.rb]: :abc:

File: spec/features/user_auth_spec.rb [hunk 14]
```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+feature "User signs up" do
+  scenario "successfully" do
+    visit root_path
+    click_on "Sign up"
```
Comment (Euraldius) [inline spec/features/user_auth_spec.rb]: Specs should also use `i18n`

File: app/views/users/registrations/edit.html.slim (hunk 8 above)
Comment (Euraldius) [inline app/views/users/registrations/edit.html.slim]: Added

File: config/environments/development.rb [hunk 15]
```diff
@@ -39,5 +39,5 @@
   # Raises error for missing translations
   config.action_view.raise_on_missing_translations = true
 
-  config.action_mailer.default_url_options = { host: "localhost:9000" }
+  config.action_mailer.default_url_options = { host: "localhost", port: 9000 }
```
Comment (jasonramirez) [inline config/environments/development.rb]: Style choice, or does this in fact have value?

Comment (Euraldius) [inline config/environments/development.rb]: Style choice :)

File: config/locales/devise.en.yml [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+      confirmation_instructions:
+        subject: "Confirmation instructions"
+      reset_password_instructions:
+        subject: "Reset password instructions"
+      unlock_instructions:
+        subject: "Unlock instructions"
+    omniauth_callbacks:
+      failure: "Could not authenticate you from %{kind} because \"%{reason}\"."
+      success: "Successfully authenticated from %{kind} account."
+    passwords:
```
Comment (jasonramirez) [inline config/locales/devise.en.yml]: A great way to keep YML files readable is to do 

```
thing: >
  Super duper long text.....
```

We can do this in a different PR

**Comment by jasonramirez** [comment]
Small question, comment for a new PR I believe, and a :hamburger:

File: config/locales/devise.en.yml (hunk 16 above)
Comment (Euraldius) [inline config/locales/devise.en.yml]: Makes sense!
